### PR TITLE
Demo improvements

### PR DESCRIPTION
**Pull request by teo-milea**
## Description
There are multiple changes meant to improve the flow of a future helm demo:
* parallelization of `file.iter_lines()` and processing
* ~new policy `ExecuteOncePerCombination`~ -> moved to #8160
* UX improvements for jobs 
* fixed problem with logs for subjobs

## How has this been tested?
* notebooks under the helm directory

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/service/action/action_service.py [hunk 1]
```diff
@@ -183,9 +183,11 @@ def _user_code_execute(
         code_item: UserCode,
         kwargs: Dict[str, Any],
     ) -> Result[ActionObjectPointer, Err]:
-        filtered_kwargs = code_item.input_policy.filter_kwargs(
+        input_policy = code_item.input_policy
+        filtered_kwargs = input_policy.filter_kwargs(
             kwargs=kwargs, context=context, code_item_id=code_item.id
         )
+        code_item.input_policy = input_policy
```
Comment (koenvanderveen) [inline packages/syft/src/syft/service/action/action_service.py:191]: why is this needed?

Comment (teo-milea) [reply packages/syft/src/syft/service/action/action_service.py:191]: if you check the UserCode class, you can see that the input policy is actually a property that also has a setter method, so in order to save the state we need to do this. This is exactly the same mechanism as the one for the output policy

**Comment by koenvanderveen** [comment]
great job!
